presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -277,6 +277,7 @@ public class FeaturesConfig
     private boolean useDefaultsForCorrelatedAggregationPushdownThroughOuterJoins = true;
     private boolean mergeDuplicateAggregationsEnabled = true;
     private boolean fieldNamesInJsonCastEnabled;
+    private boolean legacyJsonCast;
```

Comment:
It's not clear to me that this should default to false or even be considered legacy behavior.  For sure making the content lower case is a bug and should be fixed for all cases. But for the keys, quoting in json is not evidence of trying to preserve the case, it's just how you define string types, so it's not clear one way or the other to me with regards to treating them as quoted or unquoted identifiers.

Reply:
never mind, i think i misunderstood what was happening here. it's when the row fields are quoted that this matching is case sensitive, not when the json fields are in quotes.  However, I still think it's worth considering the default value, as this is a change that will break people's production deployments. IMO, we should leave the value as true at first to leave people time to migrate/update their configurations with this new property.